### PR TITLE
T64 DMR

### PR DESCRIPTION
**Pull request by David-Stormwell-TGCM**
## About The Pull Request
This PR adds a new gun, the T64 D.M.R, Designated Marksman Rifle!
Using it's own DMR Bullets, Magazines, and a new stock "Which doesn't effect the gun currently and is only there to make the gun look complete" The T64 Lacks a Burst fire, or any BFA attachments, But it does however have a automatic mode. And the T64 also has a new sound for it as well.

## Why It's Good For The Game
We've been talking about adding a DMR or something like this for a very long time now, and now with Kuro's and MJP's whining boosting me, I've made it with Kuro's sprites for it and MJP's Balance Document.

Good for Marine variety and more options for support weapons, currently it's a little shit vs the pulse rifle, but with some testing we will see how to buff or nerf areas of it. 

## Changelog
:cl:
add: T64 DMR
add: 10x27 Caliber ammo for the DMR 
soundadd: A new firing sound for the DMR
imageadd: Gun Icons and magazine icons for it.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/updated_projectiles/guns/rifles.dm [hunk 1]
```diff
@@ -103,7 +151,7 @@
 	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_AMMO_COUNTER|GUN_LOAD_INTO_CHAMBER
 
 	burst_delay = 0.2 SECONDS
-	accuracy_mult = 1.5
```
Comment (David-Stormwell-TGCM) [inline code/modules/projectiles/updated_projectiles/guns/rifles.dm]: ah fuck

**Comment by Rohesie** [comment]
There's a rogue T12 icon there.
Other than that, it looks good!

**Comment by Kuro-ai** [comment]
speed merge or riot

**Comment by Pariah919** [comment]
make it

	name = "\improper T-64 designated marksman rifle"
not
	name = "\improper T64 designated marksman rifle"

**Comment by InterroLouis** [comment]
The magazines don't fit in magazine pouches. Make it a child of the rifle mag or add it to the whitelist please. Or weight exemption, or whatever we use to let mag pouches hold magazines.

File: code/datums/gamemodes/distress.dm [hunk 2]
```diff
@@ -421,6 +423,7 @@
 						/obj/item/ammo_magazine/revolver = round(scale * 20),
 						/obj/item/ammo_magazine/smg/m39 = round(scale * 30),
 						/obj/item/ammo_magazine/rifle = round(scale * 25),
+						/obj/item/ammo_magazine/standard_dmrmag = round(scale * 25),
```
Comment (Rohesie) [inline code/datums/gamemodes/distress.dm]:
```suggestion
						/obj/item/ammo_magazine/standard_dmr = round(scale * 25),
```
Change the name to follow the convention pattern.

**Comment by Pariah919** [comment]
you forgot to make the mags fit in armor. Make the mags a child of rifle magazines or put webbing storage bypasses

**Comment by Isy232** [comment]
Even as a xeno main, this gun is underwhelming. Could use a bit of a buff.

File: code/game/objects/items/storage/belt.dm [hunk 3]
```diff
@@ -263,7 +263,7 @@
 		/obj/item/flashlight/flare,
 		/obj/item/explosive/grenade/flare,
 		/obj/item/ammo_magazine/rifle,
-		/obj/item/ammo_magazine/standard_dmrmag,
+        /obj/item/ammo_magazine/rifle/standard_dmrmag,
```
Comment (Rohesie) [inline code/game/objects/items/storage/belt.dm]: Indentation.

File: code/datums/gamemodes/distress.dm [hunk 4]
```diff
@@ -423,7 +423,7 @@
 						/obj/item/ammo_magazine/revolver = round(scale * 20),
 						/obj/item/ammo_magazine/smg/m39 = round(scale * 30),
 						/obj/item/ammo_magazine/rifle = round(scale * 25),
-						/obj/item/ammo_magazine/standard_dmrmag = round(scale * 25),
+						/obj/item/ammo_magazine/rifle/standard_dmrmag = round(scale * 25),
```
Comment (Rohesie) [inline code/datums/gamemodes/distress.dm]: Could you change the name to `/obj/item/ammo_magazine/rifle/standard_dmr` instead? The mag part is already in the name.

Comment (David-Stormwell-TGCM) [reply code/datums/gamemodes/distress.dm]: Alright then

File: code/modules/projectiles/updated_projectiles/ammo_datums.dm [hunk 5]
```diff
@@ -448,6 +448,18 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	damage = 30
 	penetration = 20
 
+/datum/ammo/bullet/rifle/standard_dmrammo
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/ammo_datums.dm]:
```suggestion
/datum/ammo/bullet/rifle/standard_dmr
```

**Comment by Pariah919** [comment]
you forgot to make the slowdown 0.5 to 0.75 david

**Comment by BurgerLUA** [comment]
It's a good weapon but it has absolute shit tier ammo management. 15 round mags is pretty small and I always seem to run out as a marine. The regular assault rifle that marines get just seems better for battles.

**Comment by Rohesie** [comment]
Small mag is good. This is a designated marksman rifle, not a spray and pray gun.

**Pull request merged**